src/components/DataTable/DataTable.js

```diff
@@ -173,10 +173,31 @@ export default class DataTable extends React.Component {
       isSortHeader: sortHeaderKey === header.key,
       // Compose the event handlers so we don't overwrite a consumer's `onClick`
       // handler
-      onClick: composeEventHandlers([this.handleSortBy(header.key), onClick]),
+      onClick: composeEventHandlers([
+        this.handleSortBy(header.key),
+        onClick
+          ? this.handleOnHeaderClick(onClick, header.key, sortDirection)
```

Comment:
Do you want to gather sortkey/sortdirection in an object here, too?

Reply:
Like this?
```
this.handleOnHeaderClick(onClick, {
  sortHeaderKey: header.key,
  sortDirection,
});```

Reply:
Yes 👍